pulsar-functions/worker/src/test/java/org/apache/pulsar/functions/worker/rest/api/v2/SinkApiV2ResourceTest.java

```diff
@@ -0,0 +1,905 @@
+        return resource.deregisterFunction(
+            tenant,
+            namespace,
+                sink,
+            null);
+    }
+
+    @Test
+    public void testDeregisterNotExistedSink() {
+        when(mockedManager.containsFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(false);
+
+        Response response = deregisterDefaultSink();
+        assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+        assertEquals(new ErrorData("Function " + sink + " doesn't exist").reason, ((ErrorData) response.getEntity()).reason);
+    }
+
+    @Test
+    public void testDeregisterSinkSuccess() throws Exception {
+        when(mockedManager.containsFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(true);
+
+        RequestResult rr = new RequestResult()
+            .setSuccess(true)
+            .setMessage("source deregistered");
+        CompletableFuture<RequestResult> requestResult = CompletableFuture.completedFuture(rr);
+        when(mockedManager.deregisterFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(requestResult);
+
+        Response response = deregisterDefaultSink();
+        assertEquals(Status.OK.getStatusCode(), response.getStatus());
+        assertEquals(rr.toJson(), response.getEntity());
+    }
+
+    @Test
+    public void testDeregisterSinkFailure() throws Exception {
+        when(mockedManager.containsFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(true);
+
+        RequestResult rr = new RequestResult()
+            .setSuccess(false)
+            .setMessage("source failed to deregister");
+        CompletableFuture<RequestResult> requestResult = CompletableFuture.completedFuture(rr);
+        when(mockedManager.deregisterFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(requestResult);
+
+        Response response = deregisterDefaultSink();
+        assertEquals(Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+        assertEquals(new ErrorData(rr.getMessage()).reason, ((ErrorData) response.getEntity()).reason);
+    }
+
+    @Test
+    public void testDeregisterSinkInterrupted() throws Exception {
+        when(mockedManager.containsFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(true);
+
+        CompletableFuture<RequestResult> requestResult = FutureUtil.failedFuture(
+            new IOException("Function deregisteration interrupted"));
+        when(mockedManager.deregisterFunction(eq(tenant), eq(namespace), eq(sink))).thenReturn(requestResult);
+
+        Response response = deregisterDefaultSink();
+        assertEquals(Status.INTERNAL_SERVER_ERROR.getStatusCode(), response.getStatus());
+        assertEquals(new ErrorData("Function deregisteration interrupted").reason, ((ErrorData) response.getEntity()).reason);
+    }
+
+    // Source Info doesn't exist. Maybe one day they might be added
+    //
+    // Get Function Info
+    //
+
+    /*
+    @Test
+    public void testGetFunctionMissingTenant() throws Exception {
```

Comment:
do we need the commented out code?

Reply:
In the next pr, we will switch to returning SourceConfig/SinkConfig as getInfo for sources/sinks instead of functionConfig. These tests which are currently copied over from FunctionApiV2ResourceTest will then be resurrected.

Reply:
ok